setup.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Check if OpenCV is installed
+echo "Checking if OpenCV is installed..."
+if pkg-config --exists opencv4 sdl2; then
+    echo "OpenCV is installed"
+else
+    echo "OpenCV is not installed"
+    echo "Installing OpenCV..."
+
+    # Detect OS and install dependencies accordingly
+    if grep -qi "debian" /etc/os-release; then
+        echo "Detected Debian-based system"
+        sudo apt-get update -y
+        sudo apt-get install libopencv-dev libsdl2-2.0-0 libsdl2-image-dev libsdl2-dev -y
```

Comment:
add a line to install 'build-essential' package (already mentioned in #129 ) for ubuntu as well as similar command for other OSs